src/libraries/Microsoft.PowerFx.Interpreter/Functions/SetFunction.cs

```diff
@@ -80,6 +78,17 @@ public override void CheckSemantics(TexlBinding binding, TexlNode[] args, DType[
                         return;
                     }
 
+                    if (arg1.IsAggregate)
+                    {
+                        var record = arg1.ToRecord();
+
+                        if (record.GetAllNames(DPath.Root).Any(name => name.Type.IsExpandEntity))

```

Comment:
>if (record.GetAllNames(DPath.Root).Any(name => name.Type.IsExpandEntity)) [](http://example.com/codeflow?start=24&length=73)

Can we encapsulate this as a helper on RecordValue? 
We may need to reuse this in other places.